src/components/RelicModal.jsx

```diff
@@ -331,174 +843,481 @@ export default function RelicModal(props) {
+                            }
+                            setButton2low(low)
+                            setButton2mid(mid)
+                            setButton2high(high)
+                            relicForm.setFieldValue('substatValue2', 0)
+                            setTrueStats({ substat2: 0 })
+                            setHide2('inline')
+                          }
+                        } else {
+                          relicForm.setFieldValue('substatValue2', undefined)
+                          setTrueStats({ substat2: 0 })
+                          setButton2low(undefined)
+                          setButton2mid(undefined)
+                          setButton2high(undefined)
+                          setHide2('inline')
+                        }
+                      }}
+                    />
+                  </Form.Item>
+
+                  <Form.Item size="default" name="substatValue2">
+                    <InputNumberStyled controls={false} precision={precision2} />
+                  </Form.Item>
+                </Flex>
+
+                <Flex gap={10}>
+                  <Form.Item size="default" name="substatType3">
+                    <Select
+                      showSearch
+                      allowClear
+                      style={{
+                        width: 200,
+                      }}
+                      placeholder="Substat"
+                      maxTagCount="responsive"
+                      options={substatOptions}
+                      onChange={(e) => {
+                        const stat = relicForm.getFieldValue('substatType3')
+                        if (stat) {
+                          if (stat == Constants.Stats.SPD) {
+                            setPrecision3(0)
+                            setHide3('none')
+                            setButton3low(2)
+                            setButton3mid(undefined)
+                            setButton3high(3)
+                            relicForm.setFieldValue('substatValue3', 0)
+                            setTrueStats({ substat3: 0 })
+                          } else {
+                            let low = SubStatValues[stat][5].low
+                            let mid = SubStatValues[stat][5].mid
+                            let high = SubStatValues[stat][5].high
+                            if ([Constants.Stats.HP, Constants.Stats.DEF, Constants.Stats.ATK].includes(stat)) {
+                              setPrecision3(0)
+                              low = Math.floor(low)
+                              mid = Math.floor(mid)
+                              high = Math.floor(high)
+                            } else {
+                              setPrecision3(1)
+                              low = Math.floor(10 * low) / 10
+                              mid = Math.floor(10 * mid) / 10
+                              high = Math.floor(10 * high) / 10
```

Comment:
Fyi floors are dangerous with multiplication/division and can lead to unexpected results - For example:

<img width="176" alt="image" src="https://github.com/fribbels/hsr-optimizer/assets/7908525/3f1ef8dc-4383-4eb2-a225-27259d9bf464">

Reply:
oh, I figured that was fine since I saw that method being used in utils.truncate10ths later on

Reply:
I try to always call Utils.precisionRound() on any numbers that go through multiplication/division before applying any floor/ceils on them, as precisionRound gets rid of any floating point messiness within a few decimal points. There's probably ones that I missed too.